tests/impls/vis_sal_classifier/test_occlusion_classifier.py

```diff
@@ -0,0 +1,101 @@
+from unittest import TestCase
+
+import numpy as np
+
+from xaitk_saliency.impls.vis_sal_classifier.occlusion_scoring import OcclusionScoring
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+
+
+class TestOcclusionBasedScoring (TestCase):
+
+    def test_init_(self) -> None:
+        """
+        Test if implementation is usable.
+        """
+        impl = OcclusionScoring()
+        assert impl.is_usable() and isinstance(impl, ImageClassifierSaliencyMapGenerator)
+
+# Below test case does pass but we should have more discussion
+# regarding support for this
+
+#     def test_multiple_images(self) -> None:
+#         """
+#         Test if implementation supports multiple images.
+#         """
+#         impl = OcclusionScoring()
+#         # Three original images with two classes
+#         image_confs_1_class_ = np.random.rand(3, 2)
+#         # Three image confidences after perturbing with six masks and two classes
+#         pertb_confs_1_class_ = np.random.rand(6, 3, 2)
+#         # Six perturbation masks for and three images
+#         mask_confs_1_class_ = np.random.randint(low=0, high=2, size=(3, 6, 10, 10), dtype='int')
+#         # Final saliency map for two classes and 3 images of height and width 10px
+#         sal = impl.generate(image_confs_1_class_, pertb_confs_1_class_, mask_confs_1_class_)
+#         # Final saliency map for two classes and 3 images of height and width 10px
+#         assert sal.shape == (2, 3, 10, 10)
+
+    def test_1class_scores(self) -> None:
+        """
+        Test basic scoring with a single class for broadcasting sanity check.
+        """
+        impl = OcclusionScoring()
+        # Three Pertrubation masks of height and width 10px for 1 class
+        image_confs_1_class_ = np.random.rand(1)
+
+        pertb_confs_1_class_ = np.random.rand(3, 1)
+        mask_confs_1_class_ = np.random.randint(low=0, high=2, size=(3, 10, 10), dtype='int')
+
+        sal = impl.generate(image_confs_1_class_, pertb_confs_1_class_, mask_confs_1_class_)
+        assert sal.shape == (1, 10, 10)
```

Comment:
It might be good to generate expected heatmaps as images or npz files and save those within the `tests/data/` directory for loading and comparison here to ensure results don't regress moving forward.

Reply:
Agreed! I added a condition for checking equality with known values in the next testing function.